source/CMakeLists.txt

```
@@ -0,0 +1,74 @@
+#
+# make sure to set env. variables first:
+#	OPENSSL_ROOT_DIR - points to OpenSSL installation
+#		prefix with desired version
+#
+# to run performance test I suggest to build binaries outside of soruces.
+# run cmake as follows:
+#	cmake -S . -B ./build
+# option -B defines location of project files and resulting binaries.
+#
+# then you can build tools using cmake:
+#	cmake --build ./build
+#
+#
+
+cmake_minimum_required(VERSION 3.10)
+project(perf-tools)
+
+add_library(perf perflib/perfhelper.c perflib/perfsslhelper.c perflib/threads.c
+	perflib/time.c perflib/getopt.c perflib/basename.c perflib/strcasecmp.c)
+
+#
+# Using PRIVATE instead of PUBLIC will will break build with 1.1.1 version
+# on Windows. This glitch is specific to 1.1.1 3.x versions do build fine.
+# So let's keep PUBLIC here.
+#
+include_directories(PUBLIC ${OPENSSL_INCLUDE_DIR} . )
+
+find_package(Threads)
+link_libraries(${CMAKE_THREAD_LIBS_INIT})
+
+find_package(OpenSSL)
```

I suggest making some requirements:

```suggestion
find_package(OpenSSL REQUIRED)
```